x/staking/types/msg.go

```diff
@@ -50,6 +52,10 @@ func NewMsgCreateValidator(
 	description Description, commission CommissionRates, minSelfDelegation sdk.Int,
 ) MsgCreateValidator {
 
+	if _, ok := pubKey.(bls12_381.PubKeyBls); !ok && pubKey != nil {
+		panic(fmt.Sprintf("Attempt to create validator with non bls based public key is invalid!\n"))
```

Comment:
I don't know if this should be a panic. Should we not just reject the msg when the MsgCreateValidator is handled?

Reply:
It does not need to be a panic, it is quite useful when making the switch to find out when code has tried to create an incorrect validator. I can remove it if you like?

Reply:
No, that's fine. I agree it is useful when switching to keep this.